### DISCUSSION TITLE
GraphQL queries not quite what I expected

### DISCUSSION THREAD

**Discussion by dodanex** [appwrite/appwrite#5435] [General]
Hello,

First I like to thank the community for this amazing product. I'm new to Appwrite and I'm trying to compare it with other GraphQL tools like Hasura to see if it can be a great alternative to build apps, so here is my feedback:

Using Appwright GraphQL I found out that it's not quite what I expected.

This query returns all the fields and relationships but it doesn't allow for custom field selection:

```gql
query {
    databasesListDocuments(
        databaseId: "dev",
        collectionId: "articles"
    ) {
        total
        documents {
            _id
            data
        }
    }
}
```

I was expecting the query api to be more cleaner, maybe like this:

```gql
query {
    databasesListDocuments(
        databaseId: "dev",
        collectionId: "articles"
    ) {
        _id
        title
        description
        user {
       	  _id
       	  name
          email
        }
    }
}
```

Or at least "data" to support an object with custom fields:

```gql
query {
    databasesListDocuments(
        databaseId: "dev",
        collectionId: "articles"
    ) {
        total
        documents {
            _id
            data {
                title
                description
                user {
                	_id
                	data {
                		name
                		email
                	}
                }
            }
        }
    }
}
```
Error "message": "Field \"data\" of type \"String\" must not have a sub selection.",

(the "data" field it's kinda' bad idea though...)

Is there any plan in the near future to support custom data fields in GraphQL ?

Another think I want to ask - What about multiple mutations at once?
Any plans to implement `databasesCreateManyDocuments, databasesUpdateManyDocuments, databasesDeleteManyDocuments`, so that we can handle arrays instead of single documents?

**Comment by joeyouss** [comment]
Hi
Thank you for your feedback. These suggestions are valuable. For multiple mutations, check this once: https://github.com/appwrite/appwrite/issues/3051 and https://github.com/appwrite/awesome-appwrite/pull/489
